### DISCUSSION TITLE
Add SQL Server Datasource

### DISCUSSION THREAD

**Discussion by ytrewq17** [Budibase/budibase#10823] [Help] [answered]
Hi everyone, I want to add a SQL Datasource to my APP but I'm getting this error

`Error updating datasource schema : ConnectionError: Logon failed for login 'XXXXXXXXXX' due to trigg`

Any idea how to solve it? Thanks

**Comment by ConorWebb96** [comment]
Hey @ytrewq17,

Can you send through a screenshot of the settings you are using to try and connect to your sql data source? Below is how mine looks if this is helpful. 
<img width="622" alt="Screenshot 2023-06-08 at 07 42 59" src="https://github.com/Budibase/budibase/assets/126772285/ac848171-4cb6-4c86-8b57-8975c86c7888">

Judging from the error it seems like you either have the wrong hosting details or haven't used the correct login/password for the database.

As I believe this is a duplicate question, there is also a response from the discord community discussions here; https://discord.com/channels/733030666647765003/1116034750570844191/1116152111495913612

I hope this helps.

**Reply by ytrewq17** [reply, marked as answer]
Thank you for answering. 
It was a firewall issue. To allow Budibase to connect to any data source you have to whitelist IP addresses in your firewall.
34.246.43.190
3.252.124.91
34.249.190.25
34.254.158.174
Once these IP addresses were allowed on the firewall, Budibase Cloud was able to access to the database server correctly.

[https://docs.budibase.com/docs/whitelisting](url)

**Reply by ConorWebb96** [reply]
Thanks for providing an answer on how you resolved this! Should help others who come against this issue in the future!
